src/iothread.c

```diff
@@ -42,6 +46,49 @@ static inline void sendPendingClientsToMainThreadIfNeeded(IOThread *t, int check
     }
 }
 
+/* When moving a client from IO thread to main thread we may need to update
+ * some of its variables as they are duplicated to avoid contention with main
+ * thread.
+ * For now this is valid only for master or slave clients. */
+void updateClientDataFromIOThread(client *c) {
+    serverAssert(c->tid != IOTHREAD_MAIN_THREAD_ID &&
+                 c->running_tid == IOTHREAD_MAIN_THREAD_ID);
+
+    if (c->io_last_ack_time / 1000 > c->repl_ack_time) {
```

Comment:
what about like the following to avoid normal users also having to go through so many branch check.
```
if (c->flags & CLIENT_SLAVE) {
} else if (c->flags & CLIENT_MASTER) {
}
```